tests/js/chartAuthoringTests.js

```diff
@@ -155,8 +159,50 @@ https://raw.githubusercontent.com/fluid-project/chartAuthoring/master/LICENSE.tx
         jqUnit.assertDeepEq("The reset has an aria-controls attribute properly referencing the form, pie, legend and total", resetAriaControlsAttr, dataEntryFormlId + " " + legendTableId + " " + pieChartPieId + " " + dataEntryFormTotalId);
     };
 
+    // Verify that the updateDataEntryPanel function updates the UI
+    floe.tests.chartAuthoringTester.verifyUpdateDataEntryPanel = function (that) {
+        that.updateDataEntryPanelFromDataSet(floe.tests.chartAuthoring.updateDataSet);
+
+        var dataEntries = that.chartAuthoringInterface.dataEntryPanel.locate("dataEntry");
+
+        var dataEntryLabelSelector = that.chartAuthoringInterface.dataEntryPanel.dataEntry.options.selectors.label;
+        var dataEntryValueSelector = that.chartAuthoringInterface.dataEntryPanel.dataEntry.options.selectors.value;
+
+        dataEntries.each(function (idx) {
+
+            // Confirm that each entry in updateDataSet has a matching dataEntry
+            // in the panel UI, or that the "matching" dataEntry is blank when
+            // the dataset is shorter than the number of possible data entries
+
+            var expectedData = idx < floe.tests.chartAuthoring.updateDataSet.length ? floe.tests.chartAuthoring.updateDataSet[idx] : expectedData = {label: "", value: ""};
+
+            jqUnit.assertEquals("Testing updated dataEntryPanel UI - displayed label at position " + idx + " matches updated data set label", $(this).find(dataEntryLabelSelector).val(), String(expectedData.label));
```

Comment:
We should have a chat about how we can clean this up - similar to the main strategy comment above about how to deal with links to data at the D3 level, it's a reasonable responsibility for a component to store a data structure which indirects from data onto DOM ids responsible for that data (most likely allocated using fluid.allocateSimpleId). Whilst we can't explicitly iterate over dynamicComponents, we can have them "report for duty on creation" by broadcasting onCreate listeners to them via distributeOptions, which will allow us to build up the data structure above.